build.gradle

```diff
@@ -71,3 +71,11 @@ subprojects {
     }
   }
 }
+
+task installGitHook(type: Copy) {
+  from new File(rootProject.rootDir, 'pre-commit')
+  into { new File(rootProject.rootDir, '.git/hooks') }
+  fileMode 0777
+}
```

Comment:
This makes me a little nervous. I'm not hugely enthusiastic about Gradle playing with version control, although I can see why it's necessary. I would be happier if there was a check in place to see if the `pre-commit` hook needed to be added rather than just overwriting whatever is already there.

Additionally: Mode `0777` is a security vulnerability. The `7` at the end means that any other user on the system can overwrite this file with arbitrary code and have you execute it under your uid next time you commit. I'd expect mode `0755` to be sufficient (or perhaps even just `0700`).